src/java.base/share/native/launcher/main.c

```diff
@@ -34,6 +34,14 @@
 #include "jli_util.h"
 #include "jni.h"
 
+#ifdef UNDEFINED_BEHAVIOR_SANITIZER
+// Override weak symbol exposed by UBSan to override default options. This is called by UBSan
+// extremely early during library loading, before main is called.
+JNIEXPORT const char* __ubsan_default_options() {
```

Comment:
Why would this need `JNIEXPORT`? This is not a JNI function.

Reply:
Ugh, apparently resolving doesn't add the comment. I thought it did...

So answering, JNIEXPORT is needed so that the symbol is exported and the linker doesn't remove it. It adds __attribute__((visibility("default")).

Reply:
Okay. The limited documentation on these `___xxx_default_options` functions didn't show the need for any export/visibility attributes.